### DISCUSSION TITLE
【web】PAG动画执行卡顿

### DISCUSSION THREAD

**Discussion by vicki2024** [Tencent/libpag#2406] [Q&A] [answered]
如何拿到PAG动画性能参数，如帧率，各阶段（初始化、执行）耗时，内存变化等

**Comment by vicki2024** [comment]
web端性能分析工具&移动端的真机性能分析 App有计划开源吗？

**Reply by kevingpqi123** [reply, marked as answer]
关于 PAG 文件的性能，可以直接在桌面端通过 PAGViewer 查看，对于移动端或 Web 端，如果你们需要自己开发就可以，这里不是很复杂，使用 PAGPlayer、PAGSurfac、PAGFile 的相关接口自己控制播放进度，统计 PAGPlayer flush 的耗时，这就是单帧耗时
